extensions/amp-auto-ads/0.1/attributes.js

```diff
@@ -29,26 +30,26 @@ const NON_DATA_ATTRIBUTE_WHITELIST = {
 };
 
 /**
- * @param {!Object} configObj
- * @return {!Object<string, string>}
+ * @param {!JsonObject} configObj
+ * @return {!JsonObject<string, string>}
  */
 export function getAttributesFromConfigObj(configObj) {
   if (!configObj['attributes']) {
-    return {};
+    return dict();
   }
   if (!isObject(configObj['attributes']) || isArray(configObj['attributes'])) {
```

Comment:
Since this is a not, it is actually sound. Unfortunately it is essentially impossible to detect the correct type at runtime. We do have cases that definitely get it now.
My changes for now, try to limit themselves to cases that the type system can detect.